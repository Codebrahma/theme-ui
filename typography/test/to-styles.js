import React from 'react'
import toStyles from '../src/to-styles'
import theme from 'typography-theme-wordpress-2016'

test('converts typography theme to styles object', () => {
  const styles = toStyles(theme)
  expect(typeof styles).toBe('object')
  expect(styles).toMatchSnapshot()
})
