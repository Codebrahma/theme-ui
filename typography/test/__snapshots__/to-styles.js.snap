// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts typography theme to styles object 1`] = `
Object {
  "*": Object {
    "boxSizing": "inherit",
  },
  "*:after": Object {
    "boxSizing": "inherit",
  },
  "*:before": Object {
    "boxSizing": "inherit",
  },
  "@media only screen and (max-width:480px)": Object {
    "blockquote": Object {
      "marginLeft": "-21px",
      "marginRight": 0,
      "paddingLeft": "15px",
    },
    "ul,ol": Object {
      "marginLeft": "28px",
    },
  },
  "a": Object {
    "boxShadow": "0 1px 0 0 currentColor",
    "color": "#007acc",
    "textDecoration": "none",
  },
  "a:hover,a:active": Object {
    "boxShadow": "none",
  },
  "abbr": Object {
    "borderBottom": "1px dotted hsla(0,0%,0%,0.5)",
    "cursor": "help",
  },
  "abbr[title]": Object {
    "borderBottom": "1px dotted hsla(0,0%,0%,0.5)",
    "cursor": "help",
    "textDecoration": "none",
  },
  "acronym": Object {
    "borderBottom": "1px dotted hsla(0,0%,0%,0.5)",
    "cursor": "help",
  },
  "address": Object {
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "b": Object {
    "fontWeight": 700,
  },
  "blockquote": Object {
    "borderLeft": "5px solid hsla(0,0%,0%,0.9)",
    "color": "hsla(0,0%,0%,0.59)",
    "fontSize": "19.2179909437029px",
    "fontStyle": "italic",
    "lineHeight": "28px",
    "marginBottom": "28px",
    "marginLeft": "-28px",
    "marginRight": "28px",
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": "22px",
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "blockquote *:last-child": Object {
    "marginBottom": 0,
  },
  "blockquote > :last-child": Object {
    "marginBottom": 0,
  },
  "blockquote cite": Object {
    "color": "hsla(0,0%,0%,0.9)",
    "fontSize": "16px",
    "fontWeight": 400,
    "lineHeight": "28px",
  },
  "blockquote cite:before": Object {
    "content": "\\"â€” \\"",
  },
  "body": Object {
    "MozFontFeatureSettings": "\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\"",
    "WebkitFontFeatureSettings": "\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\"",
    "color": "hsla(0,0%,0%,0.9)",
    "fontFamily": "'Merriweather','Georgia',serif",
    "fontFeatureSettings": "\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\"",
    "fontKerning": "normal",
    "fontWeight": 400,
    "msFontFeatureSettings": "\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\"",
    "wordWrap": "break-word",
  },
  "code": Object {
    "fontSize": "13.6px",
    "lineHeight": "28px",
  },
  "dd": Object {
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "dl": Object {
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "dt": Object {
    "fontWeight": 700,
  },
  "fieldset": Object {
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "figure": Object {
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "form": Object {
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "h1": Object {
    "color": "inherit",
    "fontFamily": "Montserrat,sans-serif",
    "fontSize": "40px",
    "fontWeight": 900,
    "lineHeight": 1.1,
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
    "textRendering": "optimizeLegibility",
  },
  "h1,h2,h3,h4,h5,h6": Object {
    "marginTop": "56px",
  },
  "h2": Object {
    "color": "inherit",
    "fontFamily": "'Merriweather','Georgia',serif",
    "fontSize": "27.725793726205854px",
    "fontWeight": 900,
    "lineHeight": 1.1,
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
    "textRendering": "optimizeLegibility",
  },
  "h3": Object {
    "color": "inherit",
    "fontFamily": "'Merriweather','Georgia',serif",
    "fontSize": "23.08319849451542px",
    "fontWeight": 900,
    "lineHeight": 1.1,
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
    "textRendering": "optimizeLegibility",
  },
  "h4": Object {
    "color": "inherit",
    "fontFamily": "'Merriweather','Georgia',serif",
    "fontSize": "16px",
    "fontWeight": 900,
    "letterSpacing": "0.140625em",
    "lineHeight": 1.1,
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
    "textRendering": "optimizeLegibility",
    "textTransform": "uppercase",
  },
  "h5": Object {
    "color": "inherit",
    "fontFamily": "'Merriweather','Georgia',serif",
    "fontSize": "13.32085131842997px",
    "fontWeight": 900,
    "lineHeight": 1.1,
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
    "textRendering": "optimizeLegibility",
  },
  "h6": Object {
    "color": "inherit",
    "fontFamily": "'Merriweather','Georgia',serif",
    "fontSize": "12.154524686917982px",
    "fontStyle": "italic",
    "fontWeight": 900,
    "lineHeight": 1.1,
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
    "textRendering": "optimizeLegibility",
  },
  "hgroup": Object {
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "hr": Object {
    "background": "hsla(0,0%,0%,0.2)",
    "border": "none",
    "height": "1px",
    "marginBottom": "calc(28px - 1px)",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "html": Object {
    "boxSizing": "border-box",
    "font": "100%/1.75 'Merriweather','Georgia',serif",
    "overflowY": "scroll",
  },
  "iframe": Object {
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "img": Object {
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "maxWidth": "100%",
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "kbd": Object {
    "fontSize": "13.6px",
    "lineHeight": "28px",
  },
  "li": Object {
    "marginBottom": "calc(28px / 2)",
  },
  "li *:last-child": Object {
    "marginBottom": 0,
  },
  "li > ol": Object {
    "marginBottom": "calc(28px / 2)",
    "marginLeft": "28px",
    "marginTop": "calc(28px / 2)",
  },
  "li > p": Object {
    "marginBottom": "calc(28px / 2)",
  },
  "li > ul": Object {
    "marginBottom": "calc(28px / 2)",
    "marginLeft": "28px",
    "marginTop": "calc(28px / 2)",
  },
  "mark,ins": Object {
    "background": "#007acc",
    "color": "white",
    "padding": "1px 3px",
    "textDecoration": "none",
  },
  "noscript": Object {
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "ol": Object {
    "listStyleImage": "none",
    "listStylePosition": "outside",
    "marginBottom": "28px",
    "marginLeft": "28px",
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "ol li": Object {
    "paddingLeft": 0,
  },
  "p": Object {
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "p *:last-child": Object {
    "marginBottom": 0,
  },
  "pre": Object {
    "fontSize": "13.6px",
    "lineHeight": "28px",
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "samp": Object {
    "fontSize": "13.6px",
    "lineHeight": "28px",
  },
  "strong": Object {
    "fontWeight": 700,
  },
  "table": Object {
    "borderCollapse": "collapse",
    "fontSize": "16px",
    "lineHeight": "28px",
    "marginBottom": "28px",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
    "width": "100%",
  },
  "td,th": Object {
    "MozFontFeatureSettings": "\\"tnum\\"",
    "WebkitFontFeatureSettings": "\\"tnum\\"",
    "borderBottom": "1px solid hsla(0,0%,0%,0.12)",
    "fontFeatureSettings": "\\"tnum\\"",
    "msFontFeatureSettings": "\\"tnum\\"",
    "paddingBottom": "calc(14px - 1px)",
    "paddingLeft": "18px",
    "paddingRight": "18px",
    "paddingTop": "14px",
    "textAlign": "left",
  },
  "th": Object {
    "fontWeight": 700,
  },
  "th:first-child,td:first-child": Object {
    "paddingLeft": 0,
  },
  "th:last-child,td:last-child": Object {
    "paddingRight": 0,
  },
  "thead": Object {
    "textAlign": "left",
  },
  "ul": Object {
    "listStyle": "disc",
    "listStyleImage": "none",
    "listStylePosition": "outside",
    "marginBottom": "28px",
    "marginLeft": "28px",
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "ul li": Object {
    "paddingLeft": 0,
  },
  "ul,ol": Object {
    "marginLeft": 0,
  },
}
`;
